File: src/subscriber.cpp
#include <ros/ros.h>
#include <geometry_msgs/Pose.h> // For geometry_msgs::Pose.
#include <iomanip> // for std::setprecision and std::fixed

// A callback function. Executed each time a new Pose
// message arrives.
void callback(const geometry_msgs::Pose& msg)
{
	ROS_INFO_STREAM(std::setprecision(2) << std::fixed
		<<<"position=(" << msg.position.x << ", " << msg.position.y << ")"
		<< " orientation=(" << msg.orientation.x << ", " 
		<< msg.orientation.y << ", "<< msg.orientation.z << ", " 
		<< msg.orientation.w << ")"); 
}

int main(int argc, char const **argv)
{
	// Initialize the ROS system and become a node.
	ros::init(argc, argv, "subscriber_node");
	ros::NodeHanlde nh;

	// Create a subscriber object.
	ros::Subscriber sub = nh.subscribe("qr/pose", 1000,
		&callback); // Here we need to choose the right topic.

	// Let ROS take over.
	ros::spin();
}

File: src/subscriber.py
#!/usr/bin/env python3

import rospy
from geometry_msgs.msg import Pose

def callback(msg: Pose):
   rospy.loginfo(f" position=({round(msg.position.x, 2)}, {round(msg.position.y, 2)}"
      f" orientation=({round(msg.orientation.x, 2)}, {round(msg.orientation.y, 2)},"
      f" {round(msg.orientation.z, 2)}, {round(msg.orientation.w, 2)})")

def main():
   # Initialize the ROS system and become a node.
   rospy.init_node("subscriber_node", anonymous=True)
   
   # Create a subscriber object.
   rospy.Subscriber("qr/pose", Pose, callback)

   # Let ROS take over.
   rospy.spin()

if __name__ == '__main__':
   try:
      main()
   except rospy.ROSInterruptException:
      pass